heron/tmaster/src/cpp/manager/stateful-checkpointer.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "manager/stateful-checkpointer.h"
+#include <iostream>
+#include <sstream>
+#include <chrono>
+#include <string>
+#include "config/physical-plan-helper.h"
+#include "manager/tmaster.h"
+#include "manager/stmgrstate.h"
+#include "errors/errors.h"
+
+namespace heron {
+namespace tmaster {
+
+StatefulCheckpointer::StatefulCheckpointer(
+  std::chrono::high_resolution_clock::time_point _tmaster_start_time)
+  : tmaster_start_time_(_tmaster_start_time) {
+  // do nothing
+}
+
+StatefulCheckpointer::~StatefulCheckpointer() { }
+
+sp_string StatefulCheckpointer::GenerateCheckpointId() {
+  // TODO(skukarni) Should we append any topology name/id stuff?
+  std::ostringstream tag;
+  tag << tmaster_start_time_.time_since_epoch().count()
```

Comment:
How do we know that a checkpoint is valid for a given packing plan? Should we build this into the id or bind it some other way? We will need this to know if a checkpoint can be assigned to a redeployed topology with a different packing plan.